tests/devices/default_qubit/test_default_qubit_native_mcm.py

```diff
@@ -38,19 +38,29 @@ def validate_counts(shots, results1, results2):
     Passes if counts are too low, chosen as ``100``.
     Otherwise, fails if counts differ by more than ``20`` plus 20 percent.
     """
-    if isinstance(results1, Sequence):
-        assert isinstance(results2, Sequence)
-        assert len(results1) == len(results2)
+    if isinstance(shots, Sequence):
+        assert isinstance(results1, tuple)
+        assert isinstance(results2, tuple)
+        assert len(results1) == len(results2) == len(shots)
+        for s, r1, r2 in zip(shots, results1, results2):
+            validate_counts(s, r1, r2, batch_size=batch_size)
+        return
+
+    if batch_size is not None:
+        assert isinstance(results1, Iterable)
+        assert isinstance(results2, Iterable)
+        assert len(results1) == len(results2) == batch_size
         for r1, r2 in zip(results1, results2):
-            validate_counts(shots, r1, r2)
+            validate_counts(shots, r1, r2, batch_size=None)
         return
+
     for key1, val1 in results1.items():
         val2 = results2[key1]
         if abs(val1 + val2) > 100:
             assert np.allclose(val1, val2, rtol=20, atol=0.2)
```

Comment:
I think `atol`, `rtol` are swapped here (and two other places). Let's make the change.
```suggestion
            assert np.allclose(val1, val2, atol=20, rtol=0.2)
```